### ISSUE TITLE
Chefdk ships ruby 2.3.0 but cookstyle still uses the ruby 2.1.0 parser

### ISSUE THREAD

**Issue by kisoku** [chef/cookstyle#19]
I've been using the new squiggly heredoc in my latest cookbook code and cookstyle is constantly barfing on it.

```
libraries/lldpnotifier_service.rb:41:7: E: unexpected token tLSHFT
(Using Ruby 2.1 parser; configure using TargetRubyVersion parameter, under AllCops)
      <<~eos
```

**Comment by kisoku** [comment]
And yes I have local workarounds in place, but it would be nice if cookstyle worked out of the box with the ruby that ships with chefdk

**Comment by tas50** [comment]
@kisoku Which chef-dk release are you on?

**Comment by kisoku** [comment]
```
miyamoto [14:45:13] fastly $ chef --version
Chef Development Kit Version: 1.1.16
chef-client version: 12.17.44
delivery version: master (83358fb62c0f711c70ad5a81030a6cae4017f103)
berks version: 5.2.0
kitchen version: 1.14.2
```

**Comment by kisoku** [comment]
Here's my workaround. I shove this in a local .rubocop.yml

```
AllCops:
  TargetRubyVersion: 2.3
```

**Comment by tas50** [comment]
We might just need to require 2.3 in cookstyle and shove that into our config there.

**Comment by lamont-granquist** [comment]
community cookbooks probably still need to support ruby 2.0, certainly ruby 2.1 and barfing on the squiggly heredoc is probably a feature, not a bug.

**Comment by lamont-granquist** [comment]
to be more precise:  chef-client 12.5 is still supported by community cookbooks and it shipped with ruby 2.0 so while 12.5 is supported by community cookbooks they MUST NOT use 2.3 features like squiggly heredoc.

we only moved to ruby 2.1 in 12.8, we finally moved to 2.3 in 12.14.x

once the community cookbooks no longer support chef-client versions < 12.14.x then the default parser can be bumped to 2.3

**Comment by tas50** [comment]
I've locked Cookstyle to Ruby 2.0 for 1.4.0 since we shipped 2.0 in the 12.X chef releases. Once we EOL Chef 12 we'll be able to bump to 2.4.0 which shipped with Chef 13. Until then we're going to have to keep supporting old Ruby :(
